osu.Game/Screens/Select/Leaderboards/BeatmapLeaderboard.cs

```diff
@@ -117,9 +118,15 @@ protected override APIRequest FetchScores(CancellationToken cancellationToken)
                 return null;
             }
 
+            if (fetchRuleset.OnlineID <= 0 || fetchRuleset.OnlineID > ILegacyRuleset.MAX_LEGACY_RULESET_ID)
```

Comment:
Almost missed this one...

```suggestion
            if (fetchRuleset.OnlineID < 0 || fetchRuleset.OnlineID > ILegacyRuleset.MAX_LEGACY_RULESET_ID)
```

Launch game and view any osu! leaderboard to confirm why. Not sure whether this warrants more test coverage or not...

Reply:
In fact this check is starting to show up in so many places that it may be better to add an extension method for this rather than reimplementing all over at this point. Unsure whether that should happen here or not either, will leave to your discretion.

Reply:
Was definitely considering the extension method route while writing this. I'll look to do so, as this was a pretty huge oversight and this code is definitely in a few places.

Reply:
I've converted all usages to a property call. Seems better than an extension method (we usually don't use methods for `IsSomething`) but can convert it if you'd prefer.